crates/precompile/benches/eip2537.rs

```diff
@@ -0,0 +1,495 @@
+use ark_bls12_381::{Fq, Fr, G1Affine, G2Affine};
+use ark_ec::AffineRepr;
+use ark_serialize::CanonicalSerialize;
+use arkworks_general::{encode_field_32_bytes, random_field, random_points};
+use criterion::{measurement::Measurement, BenchmarkGroup};
+use primitives::Bytes;
+use rand::{rngs::StdRng, Rng, SeedableRng};
+use revm_precompile::bls12_381_const::{
+    FP_LENGTH, FP_PAD_BY, PADDED_FP_LENGTH, PADDED_G1_LENGTH, PADDED_G2_LENGTH,
+};
+
+// Constants for benchmarking configurations
+const RNG_SEED: u64 = 42;
+const G1_ADD_TEST_VECTORS: usize = 10;
+const G2_ADD_TEST_VECTORS: usize = 10;
+const G1_MSM_TEST_VECTORS: usize = 5;
+const G2_MSM_TEST_VECTORS: usize = 5;
+const MAX_MSM_SIZE: usize = 16;
+const PAIRING_TEST_VECTORS: usize = 5;
+const MAX_PAIRING_PAIRS: usize = 10;
+const MAP_FP_TO_G1_TEST_VECTORS: usize = 10;
+const MAP_FP2_TO_G2_TEST_VECTORS: usize = 10;
```

Comment:
This is overkill, will remove -- the rationale was that some of the operations might be too fast, so each benchmark run will cause a lot of variance. To overcome this, we can benchmark a N batches of the operation instead. But checking the CI, I see that bn128_add does not show up often in other PRs for example

Reply:
has been changed now